syntax = "proto3";

package helloworld;

// Service Definition
service Greeter {
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc SayHelloGroup (HelloRequest) returns (stream HelloReply) {}
  rpc HelloEveryone(stream HelloRequest) returns (HelloReply) {}
  rpc SayHelloOneByOne(stream HelloRequest) returns (stream HelloReply) {}
}

// Message Definition
message HelloRequest {
  string name = 1;
}

message HelloReply {
  string message = 1;
}